### PR TITLE
Clinical Trial Info Revamp

### PR DESCRIPTION
**Pull request by Dtphelan1**
# Summary
Revamps the templates for ResearchSubject and ResearchStudy with our new template approach

## New behavior
- Snippets Added: 
  - identifier
  - identifierArr, for identifiers that need to contain a list of values
  - extensionArr, which was called extension previously

## Code changes
- Removed the old `ResearchSubject` and `ResearchStudy` ejs templates, replacing them with the new JS templating approach.
- Added the snippets mentioned above
- Added tests for the identifier functions added

# Testing guidance
- Make sure all tests pass
- Ensure integration with the client works via an `npm install` pointing at the local file.

### CODE REVIEW

**Review by julianxcarter — APPROVED** [review]
This looks great to me! I think the snippet name changes make a lot of sense as well.

File: src/templates/ResearchStudyTemplate.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const { identifierArr } = require('./snippets');
+
+function siteTemplate(clinicalSiteID) {
+  return {
+    site: [
+      {
+        display: 'ID associated with Clinical Trial',
+        identifier: {
```
[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/templates/ResearchStudyTemplate.js]: No I'm just a bad developer

[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/templates/ResearchStudyTemplate.js]: Addressed.

[the remark this answers is not in the document]
Comment (Dtphelan1) [reply src/templates/ResearchStudyTemplate.js]: via an _act of god_™️

**Review by jafeltra — APPROVED** [review]
This looks really good! I didn't notice anything else and I compared to the previous version and it all looks good!

**Pull request merged**